dts/bindings/stepper/adi/adi,tmc5041.yaml

```
@@ -1,7 +1,60 @@
 # SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
 # SPDX-License-Identifier: Apache-2.0
 
-description: Analog Devices TMC5041 Stepper Motor Controller
+description: |
+  Analog Devices TMC5041 Stepper Motor Controller
+
+  Example:
+
+    #include <zephyr/dt-bindings/stepper/adi/tmc5041_reg.h>
+
+    &spi0 {
+        /* SPI bus options here, not shown */
+
+        /* Dual controller/driver for up to two 2-phase bipolar stepper motors */
+        tmc5041: tmc5041@0 {
+            compatible = "adi,tmc5041";
+            reg = <0>;
+            spi-max-frequency = <DT_FREQ_M(24)>; /* Maximum SPI bus frequency */
+
+            #address-cells = <1>;
+            #size-cells = <0>;
+
+            poscmp_enable; test_mode; lock_gconf; /* ADI TMC Global configuration flags */
+            clock-frequency = <DT_FREQ_M(16)>; /* Internal/External Clock frequency */
+
+            motor: motor@0 {
+                status = "okay";
+                reg=<0>;
```

```suggestion
                reg = <0>;
```

also fix elsewhere in the description please